snapcraft/cli/lifecycle.py

```diff
@@ -193,12 +237,18 @@ def clean(parts, step_name, **kwargs):
             step_name = "prime"
         step = steps.get_step_by_name(step_name)
 
-    if build_environment.is_host:
-        lifecycle.clean(project, parts, step)
-    else:
+    if build_environment.is_lxd:
         lxd.Project(project=project, output=None, source=os.path.curdir).clean(
             parts, step
         )
+    elif build_environment.is_host:
+        lifecycle.clean(project, parts, step)
+    else:
+        build_provider_class = build_providers.get_provider_for(
+            build_environment.provider
+        )
+        # TODO support for steps.
```

Comment:
It's ignoring parts as well. As it is in this PR, if I'm using build VMs and I run `snapcraft clean foo` or `snapcraft clean -s build` or `snapcraft clean foo -s build` it'll just toast my VM, right? Should we error instead?

Reply:
We can add that, an error that is.

Reply:
OK, I have added a temporary error message

Reply:
issue taken care of